test_conformance/api/test_api_consistency.cpp

```diff
@@ -455,13 +415,17 @@ int test_consistency_pipes(cl_device_id deviceID, cl_context context,
                            "clCreatePipe did not return CL_INVALID_OPERATION");
 
         // clGetPipeInfo
-        // Returns CL_INVALID_MEM_OBJECT since pipe cannot be a valid pipe
-        // object.
-        error = clGetPipeInfo(mem, CL_PIPE_PACKET_SIZE, sizeof(u), &u, NULL);
-        test_failure_error(
-            error, CL_INVALID_MEM_OBJECT,
-            "CL_DEVICE_PIPE_SUPPORT returned CL_FALSE but "
-            "clGetPipeInfo did not return CL_INVALID_MEM_OBJECT");
+        // Returns CL_INVALID_OPERATION if no devices in the context associated
+        // with pipe support Pipes.
+        clMemWrapper not_a_pipe =
+            clCreateBuffer(context, CL_MEM_READ_WRITE, 4, NULL, &error);
+        test_error(error, "Unable to create non-pipe buffer");
+
+        error =
+            clGetPipeInfo(not_a_pipe, CL_PIPE_PACKET_SIZE, sizeof(u), &u, NULL);
+        test_assert_error(error == CL_INVALID_OPERATION,
```

Comment:
I think this should test for `error == CL_INVALID_OPERATION || error == CL_INVALID_MEM_OBJECT`

Reply:
Yeah, I used to have both of these cases, but I think this is correct with the latest spec updates.  See: https://github.com/KhronosGroup/OpenCL-Docs/pull/418#discussion_r478742972

Reply:
Interesting, I'd missed the latest developments. Assuming we require the function to always be present in the dispatch table, then this is fine. I don't think we say that explicitly but maybe it's "obvious" since we require the function to be present and have a given behaviour.